# 1: Parsing The File

# Instructions
Open and read in la_weather.csv.
Split the data on the newline character to convert it to a list of rows.
Split each row on the comma and append each list to weather_data.
Hint
Remember to split each row on the comma (,) to get two values.



In [10]:
weather_data = []

f = open('la_weather.csv', 'r', encoding='utf-8')
data = f.read()
rows = data.split('\n')
print(rows[0:5])

for row in rows:
    split_row = row.split(",")
    weather_data.append(split_row)
print(weather_data[0:5])   

['Day,Type of Weather', '1,Sunny', '2,Sunny', '3,Sunny', '4,Sunny']
[['Day', 'Type of Weather'], ['1', 'Sunny'], ['2', 'Sunny'], ['3', 'Sunny'], ['4', 'Sunny']]


# 2: Getting A Single Column From The Data

# Instructions
Loop over each row in weather_data.
Append the second item in each row to the weather list.
When complete, weather should contain each value from the Type of Weather column.
Hint
Use the append() method to add to a list.

In [17]:
weather = []

for wd in weather_data:
    value = wd[0]
    weather.append(wd[1])
print(weather[0:10])

['Type of Weather', 'Sunny', 'Sunny', 'Sunny', 'Sunny', 'Sunny', 'Rain', 'Sunny', 'Sunny', 'Fog']


# 3: Counting The Items In A List

# Instructions
Count the number of items in weather. You can accomplish this by:
Looping over each element in weather.
Adding 1 to count for each element.
When finished, count should equal the number of items in weather.
Hint
Use a for loop to iterate over weather.

In [19]:
count = 0

for w in weather:
    count += 1
    
print(count)

366


# 4: Removing The Header

# Instructions
Slice the weather list to remove the header.
The slice should only remove the first element in the list.
Assign the slice to new_weather.
Hint
Slice from index 1 to index 366. Remember that the second index we specify needs to be greater than the maximum index in the list.



In [36]:
end = len(weather) 
print(end)
new_weather = weather[1:end]
print(new_weather[0:10])

366
['Sunny', 'Sunny', 'Sunny', 'Sunny', 'Sunny', 'Rain', 'Sunny', 'Sunny', 'Fog', 'Rain']


# 5: The In Statement

# Instructions
Use the in statement to check whether the value cat is in the list animals, and assign the result to cat_found.
Use the in statement to check whether the value space_monster is in the list animals, and assign the result to space_monster_found.
Hint
10 in animals will check whether the value 10 is in the list animals. a = 10 in animals will assign the Boolean result to the variable a.


In [57]:
animals = ["cat", "dog", "rabbit", "horse", "giant_horrible_monster"]

In [58]:
cat_found = "cat" in animals
print(cat_found)

True


In [60]:
space_monster_found = "space_monster" in animals
print(space_monster_found)

False


In [67]:
weather_types = ["Rain", "Sunny", "Fog", "Fog-Rain", "Thunderstorm", "Type of Weather"]
weather_type_found = []

for wt in weather_types:
    type_found = wt in new_weather
    weather_type_found.append(type_found)
print(weather_type_found)

[True, True, True, True, True, False]
